[18.0][FIX] stock_release_channel: delivery date

jbaudoux opened the pull request:
When the transfer is not processed on time, use now for delivery date computation

cc @simahawk

jbaudoux committed:
[FIX] stock_release_channel: delivery date

When the transfer is not processed on time, use now for delivery date computation

jbaudoux committed:
[IMP] stock_release_channel: show delivery date

Show delivery date when different from scheduled date

simahawk commented on stock_release_channel/models/stock_picking.py:
```diff
@@ -49,24 +51,44 @@ def _inverse_release_channel_id(self):
             all_moves._release_set_expected_date(new_date)
 
-    @api.depends("release_channel_id", "need_release", "partner_id")
+    @api.depends(
```
the logic looks good but I would like to move this stuff to where the triggers happen (eg: set the rel chan, change the state, release post hook, etc) as we discussed.

SilvioC2C commented on stock_release_channel/models/stock_picking.py:
```diff
@@ -49,24 +51,44 @@ def _inverse_release_channel_id(self):
+            if picking.state == "done":
+                end_preparation_date = picking.date_done
+            else:
+                end_preparation_date = max(
+                    picking.scheduled_date, fields.Datetime.now()
+                )
```
Say today it's the 01/10, the `scheduled_date` is 15/10, and this method is triggered => the picking's `delivery_date` is computed based on the `scheduled_date`.
Nothing changes on the picking for a month, so today is now 01/11. Since nothing changed, the method has not been triggered anymore since last month, and the picking's `delivery_date` is still based on the `scheduled_date`, not `datetime.now()`.
Is this expected?

jbaudoux replied:
Indeed, I need to fix that now that I store the field

simahawk replied:
If you switch to the approach I mentioned above, you won't have this issue anymore ;)

SilvioC2C commented on stock_release_channel/models/stock_picking.py:
```diff
@@ -49,24 +51,44 @@ def _inverse_release_channel_id(self):
-            if not channel or not picking.partner_id or picking.need_release:
+            if (
+                picking.state == "cancel"
+                or not picking.picking_type_code == "outgoing"
```
Is there a reason why `picking_type_code` is not included in the compute method deps?

jbaudoux replied:
it won't change

jbaudoux committed:
[IMP] stock_release_channel: delivery date fallback

Compute a delivery date when a delivery has been released without a channel.
Apply same logic than on sales by taking all possible channels.

jbaudoux committed:
fixup! [IMP] stock_release_channel: delivery date fallback

simahawk commented:
rebased

simahawk reviewed (changes requested):
Needs refactoring to ensure the date is set by proper triggers where/when needed.

jbaudoux commented:
rebased

simahawk commented:
@jbaudoux status?
